### ISSUE TITLE
🐛 [cloud_firestore] Code that works fine on iOS, does not add document on macOS

### ISSUE THREAD

**Issue by puf** [firebase/flutterfire#11007]
## Bug report

Code that adds a document to a collection on iOS, does not work when run on macOS. The Future returns by the `add()` method neither resolves nor rejects.

### Steps to reproduce

The minimal code
   ```
   floatingActionButton: FloatingActionButton(
     onPressed: () async {
       print('start!');
       await FirebaseFirestore.instance
           .collection('macTestW')
           .add({'name': 'test3'});
       print('save!');
       final test2 =
           await FirebaseFirestore.instance.collection('macTestW').get();
       test2.docs.forEach((element) {
         print('${element.id} : ${element.data()}');
       });
       print('done!');
     },
     tooltip: 'Increment',
     child: const Icon(Icons.add),
   ),
 );
```

Or alternatively without `async` and `await`:
```
print('start!');
FirebaseFirestore.instance
    .collection('macTestW')
    .add({'name': 'test3'})
    .then((e) { 
      print('save!');
      FirebaseFirestore.instance.collection('macTestW').get().then((test2) {
        test2.docs.forEach((element) {
          print('${element.id} : ${element.data()}');
        });
      });
      print('done!');
    })
    .catchError((e) {
      print("ERROR: $e");
    });
```

In both of these cases, the code only prints `start!` when run on macOS.

### Expected behavior

I expect either the `then` or the `catchError` block to be called, but neither happens. Due to the [inability to enable debug logging](https://github.com/firebase/flutterfire/issues/10995) through the Flutter SDK, it is hard to troubleshoot what is going on under the hood.

### Sample project

TODO: 

---

## Additional context

---

### Flutter doctor

Run `flutter doctor` and paste the output below:

<details><summary>Click To Expand</summary>

```
% flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.7.3, on macOS 13.3.1 22E772610a darwin-x64, locale en-US)
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
[✓] Xcode - develop for iOS and macOS (Xcode 14.2)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2021.2)
[✓] VS Code (version 1.78.2)
[✓] Connected device (2 available)
[✓] HTTP Host Availability

• No issues found!
```

</details>

---

### Flutter dependencies

Run `flutter pub deps -- --style=compact` and paste the output below:

<details><summary>Click To Expand</summary>

```
% flutter pub deps -- --style=compact
Dart SDK 2.19.2
Flutter SDK 3.7.3
store_test_mac 1.0.0+1

dependencies:
- cloud_firestore 4.7.1 [cloud_firestore_platform_interface cloud_firestore_web collection firebase_core firebase_core_platform_interface flutter meta]
- cupertino_icons 1.0.5
- firebase_core 2.13.0 [firebase_core_platform_interface firebase_core_web flutter meta]
- flutter 0.0.0 [characters collection js material_color_utilities meta vector_math sky_engine]

dev dependencies:
- flutter_lints 2.0.1 [lints]
- flutter_test 0.0.0 [flutter test_api path fake_async clock stack_trace vector_math async boolean_selector characters collection js matcher material_color_utilities meta source_span stream_channel string_scanner term_glyph]

transitive dependencies:
- _flutterfire_internals 1.3.1 [collection firebase_core firebase_core_platform_interface flutter meta]
- async 2.10.0 [collection meta]
- boolean_selector 2.1.1 [source_span string_scanner]
- characters 1.2.1
- clock 1.1.1
- cloud_firestore_platform_interface 5.14.1 [_flutterfire_internals collection firebase_core flutter meta plugin_platform_interface]
- cloud_firestore_web 3.5.1 [_flutterfire_internals cloud_firestore_platform_interface collection firebase_core firebase_core_web flutter flutter_web_plugins js]
- collection 1.17.0
- fake_async 1.3.1 [clock collection]
- firebase_core_platform_interface 4.8.0 [collection flutter flutter_test meta plugin_platform_interface]
- firebase_core_web 2.5.0 [firebase_core_platform_interface flutter flutter_web_plugins js meta]
- flutter_web_plugins 0.0.0 [flutter js characters collection material_color_utilities meta vector_math]
- js 0.6.5 [meta]
- lints 2.0.1
- matcher 0.12.13 [meta stack_trace]
- material_color_utilities 0.2.0
- meta 1.8.0
- path 1.8.2
- plugin_platform_interface 2.1.4 [meta]
- sky_engine 0.0.99
- source_span 1.9.1 [collection path term_glyph]
- stack_trace 1.11.0 [path]
- stream_channel 2.1.1 [async]
- string_scanner 1.2.0 [source_span]
- term_glyph 1.2.1
- test_api 0.4.16 [async boolean_selector collection meta source_span stack_trace stream_channel string_scanner term_glyph matcher]
- vector_math 2.1.4
```

</details>

---

**Comment by gwanhyeong** [comment]
I attached the sample project.
https://drive.google.com/file/d/1EsbouDkhf19TDMVxySFtQ7THX3sglxVD/view

**Comment by russellwheatley** [comment]
I've tested this using both async/non-async methods, testing on both Flutter versions 3.10.0 & 3.7.3. I have the same version of Xcode installed. Every time the document was added to the collection. Not sure what else could be different in the environment to have an effect. Are there any other steps to reproduce this?

**Comment by puf** [comment]
@russellwheatley I tested with OPs project, so can't see if the document was added. But I only get the `started!` print output, which is unexpected. What print output do you get?

**Comment by russellwheatley** [comment]
I got the prints expected. Example using async method:
<img width="383" alt="Screenshot 2023-05-22 at 10 04 17" src="https://github.com/firebase/flutterfire/assets/16018629/a8e74683-6b9a-4449-ad70-4525f9df5f02">

**Comment by puf** [comment]
Hmm... I wonder if writes are not allowed on @gwanhyeong's project?

**Comment by gwanhyeong** [comment]
1. I have tested it on Flutter versions 3.3.7 & 3.7.11, but it still doesn't work on macOS. Is it correct that you have tested it on macOS? @russellwheatley 
2. Here is the current Cloud Firestore security rule @puf 
```
rules_version = '2';
service cloud.firestore {
  match /databases/{database}/documents {
    match /{document=**} {
      allow read, write: if true;
    }
  }
}
```

**Comment by puf** [comment]
@russellwheatley Did you test in the macOS project that [gwanhyeong](https://github.com/gwanhyeong) provider? Or in one of your own?

@gwanhyeong Are any other Firestore operations working in this macOS project? If not, it sounds like the SDK may not be able to reach the server - which could point to a configuration problem. Once https://github.com/firebase/flutterfire/pull/10996 lands, enabling logging should also show proof of that.

**Comment by russellwheatley** [comment]
I tested the sample app, it needs "Outgoing Connections (Client)" to be checked for it to work:
<img width="812" alt="Screenshot 2023-05-25 at 08 29 34" src="https://github.com/firebase/flutterfire/assets/16018629/2a85b7f1-9de8-456d-b94f-f900dde47d7f">

Open the project in Xcode and click on the "Signing and Capabilities" tab. You'll see "App Sandbox (Debug and Profile)" appear with this checkbox. It will work once you have checked the "Outgoing Connections (Client)". If you want it to work in release mode, you will have to check the same checkboxes for "App Sandbox (Release)" as well.

Let me know if that works for you @gwanhyeong 👍

**Comment by puf** [comment]
Great find Russell. And I should've realized this right away, as I recently had to deal with these entitlements for a macOS Flutter app myself too. In fact, when I just opened that project, Xcode showed the Entitlements tab. :)

When I check that box, it shows the expected output:
```
flutter: start!
flutter: save!
flutter: done!
flutter: 23tYlBjMQpPawrzRTmMe : {name: test3}
flutter: 2XFVuVaXGJUHPs9BjXXn : {name: test3}
...
```

Assuming this fixes the problem for @gwanhyeong too, I'm not sure if there's anything we can improve for this - as the macOS support doesn't really have its own place in the docs. I hope devs in a similar situation will find the solution here.
